### PR TITLE
feat: add autotools_ls mapping

### PR DESCRIPTION
**Pull request by chrisgrieser**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -191,6 +192,7 @@ local DEFAULT_SETTINGS = {
 | Clarity                             | `clarity_lsp`                     |
 | Clojure                             | `clojure_lsp`                     |
 | CodeQL                              | `codeqlls`                        |
+| Config                              | `autotools-language-server`       |
```
Comment (williamboman) [inline README.md:195]: I'm not sure it's clear what "Config" would refer to here? Would Autoconf perhaps be better?

Comment (chrisgrieser) [reply README.md:195]: tbh, I have no idea what exactly that filetype is. I took a look at the filetypes listed at [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#autotools_ls), and just applied those. (I only use makefiles, so I am not really familiar with what the other filetypes are)

Comment (chrisgrieser) [reply README.md:195]: should I change it to `Autoconf` or leave it as `Config`?

Comment (williamboman) [reply README.md:195]: Let's change to `Autoconf`. This table is really only to allow people to search through Ctrl + F as well as some SEO so I tend to use whatever I think people would search for, not necessarily the underlying filetypes.

Comment (chrisgrieser) [reply README.md]: okay, done 👍

**Pull request merged**